kbase-extension/static/kbase/js/common/cellComponents/tabs/results/outputWidget.js

```diff
@@ -0,0 +1,80 @@
+define(['bluebird', 'jquery', 'common/html', 'common/ui', 'common/events'], function (
+    Promise,
+    $,
+    html,
+    UI,
+    Events
+) {
+    'use strict';
+
+    let tag = html.tag,
+        div = tag('div');
+
+    function OutputWidget() {
+        let container, ui;
+
+        function renderOutput(data) {
+            return new Promise((resolve) => {
+                console.log('ready to render object output with data: ', data);
+                resolve(true);
+            });
+        }
+
+        function renderLayout() {
+            const events = Events.make();
+
+            const content = ui.buildCollapsiblePanel({
+                title: 'Objects',
+                name: 'created-objects-toggle',
+                hidden: false,
+                type: 'default',
+                classes: ['kb-panel-container'],
+                body: div,
+            });
+
+            let objectContainer = div({ dataElement: 'created-objects' }, [content]);
+
+            return {
+                content: objectContainer,
+                events: events,
```

Comment:
Why are events being returned here instead of being attached to the container?

Reply:
they are attached to the container, in the doAttach method. first the layout object is build in this method (with content and events) and then in doAttach everything is piped together (line 52). i don't have a strong opinion on this flow, was just following the pattern we've used for the other widgets.

Reply:
Makes sense